### DISCUSSION TITLE
So what do you guys do when you can't sleep?

### DISCUSSION THREAD

**Discussion by ellielo** [NAIST-SE/AttractingContributors#354] [General]
Simple question. One of my friends is having a problem sleeping, well has been for quite a while and I dunno what to recommend.

**Comment by hernessteve** [comment]
Think about all the pros, remove all the cons in my mind and make myself happy. I don't try to think about something that will make me excited, because if you do that it can increase your heart rate which makes falling asleep a lot harder. So, think about something that won't prevent you from falling asleep like a nightmare, or someone who threatens you with something, (lol) and you should doze off fairly quickly.

**Comment by XelaAl** [comment]
Ever since I turned 22, I only sleep 4.5-5 hours per night. No matter how I try, I will wake up after less than 5 hours and not be able to return to sleep. I took sleeping pills, but such medications can't be used for a long time (I double-checked this info on the [Canada Drugs](https://www.canadadrugsdirect.com/) site). Currently, I'm doing yoga in the hope it will help. And thanks for sharing your experience, it is always great to learn from others.
